test/init/init_test.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+'use strict';
+
+const assert = require('chai').assert;
+const sinon = require('sinon');
+const helpers = require('yeoman-test');
+const childProcess = require('child_process');
+const inquirer = require('inquirer');
+const YeomanEnvironment = require('yeoman-environment');
+
+const polymerInit = require('../../lib/init/init');
+
+var isPlatformWin = /^win/.test(process.platform);
+
+function stripAnsi(str) {
+  let ansiRegex = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
+  return str.replace(ansiRegex, '');
+}
+
+suite('init', () => {
+  let sandbox;
+
+  function createFakeEnv() {
+    return {
+      getGeneratorsMeta: sandbox.stub(),
+      run: sandbox.stub().yields(),
+    };
+  }
+
+  setup(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  teardown(() => {
+    sandbox.restore();
+  });
+
+  suite('runGenerator', () => {
+
+    test('runs the given generator', () => {
+      const GENERATOR_NAME = 'TEST-GENERATOR';
+      let yeomanEnv = createFakeEnv();
+      yeomanEnv.getGeneratorsMeta.returns({
+        [GENERATOR_NAME]: GENERATOR_NAME,
+      });
+
+      return polymerInit.runGenerator(GENERATOR_NAME, {env: yeomanEnv}).then(() => {
+        assert.isOk(yeomanEnv.run.calledWith(GENERATOR_NAME));
+      });
+    });
+
+    test('fails if an unknown generator is requested', (done) => {
+      const UNKNOWN_GENERATOR_NAME = 'UNKNOWN-GENERATOR';
+      let yeomanEnv = createFakeEnv();
+      yeomanEnv.getGeneratorsMeta.returns({
+        'TEST-GENERATOR': 'TEST-GENERATOR',
+      });
+
+      polymerInit.runGenerator(UNKNOWN_GENERATOR_NAME, {env: yeomanEnv}).then(() => {
+        done(new Error('The promise should have been rejected by now'));
+      }).catch((err) => {
+        assert.equal(err.message, `Template ${UNKNOWN_GENERATOR_NAME} not found`);
+        done();
+      });
+    });
+
+  });
+
+  suite('promptGeneratorSelection', () => {
```

Comment:
Totally understand that all the same current behavior *could* be tested via `createSelectPrompt()` directly by making it public, but I've seen internal implementations change enough times that I just always prefer to test the public interface whenever possible even if it's a little harder to write.

Sounds like it's just a difference of opinion :) And I think you're right, with so few tests here this is more of a conceptual discussion than anything. I'm happy to make these change if you'd like me too.